mutiny-core/src/federation.rs

```diff
@@ -917,13 +1068,25 @@ where
                         log_trace!(logger, "Streamed Outcome final, returning");
                         break;
                     }
-                } else {
-                    log_debug!(
-                        logger,
-                        "Timeout reached, exiting loop for payment {}",
-                        invoice.payment_hash
-                    );
-                    break;
+                }
+                    }
+                    _ = delay_fut => {
+                    if timeout.is_none() {
+                        // TODO check sleep time to determine break
+                        log_debug!(
+                            logger,
+                            "Still waiting for stream to finish {}",
+                            invoice.payment_hash
+                        );
```

Comment:
need to add the stop channel to `federation` and check it here. Also this log is too noisy.